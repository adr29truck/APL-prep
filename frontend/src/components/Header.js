import '../Header.css';
import ArrowLeftIcon from '@material-ui/icons/ArrowLeft';
import ArrowRightIcon from '@material-ui/icons/ArrowRight';

export const Header = (props) => {
  return (<header>
    <ArrowLeftIcon onClick={props.onClickLeftArrow}/>
    <h3 className="text-center">Timeline</h3>
    <ArrowRightIcon onClick={props.onClickRightArrow}/>
  </header>)
}