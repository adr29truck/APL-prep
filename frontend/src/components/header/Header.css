header {
  display: flex;
  justify-content: space-evenly;
}
header h3 {
  margin: 4px 0;
}