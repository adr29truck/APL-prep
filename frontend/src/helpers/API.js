import axios from 'axios';

const baseUrl = () => {
  try {
    return `${window.location.protocol}//${window.location.hostname}:5000/`;
  } catch {
    return 'http://localhost:5000/';
  }
};

/**
 * Represents everythin API related
 */
export default class API {
  /**
   * Does get requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @return {object} JSON response from API request
   */
  static get(route) {
    return axios.get(baseUrl() + route)
        .then((res) => {
          return res.data;
        }).catch((error) => {
          console.log(error);
        });
  }

  /**
   * Does post requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @param {object} data - data to pass to the API
   * @return {object} JSON response from API request
   */
  static post(route, data) {
    return axios.post(baseUrl() + route, data)
        .then((res) => {
          return res.data;
        }).catch((error) => {
          console.log(error);
        });
  }
}
