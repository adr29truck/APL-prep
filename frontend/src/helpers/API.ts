/* eslint-disable no-console */
import axios, { AxiosResponse, AxiosError, AxiosRequestConfig } from 'axios';

const baseUrl = () => {
  try {
    return `${window.location.protocol}//${window.location.hostname}:5000/`;
  } catch {
    return 'http://localhost:5000/';
  }
};

const axiosConfig: AxiosRequestConfig = {
  headers: {
    'content-Type': 'application/json',
    Accept: '/',
    'Cache-Control': 'no-cache',
    // Authorization: localStorage.getItem('jwt'),
    // FIXME: IM BROKEN
    Authorization:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE2MTEyMzk0NjUsImV4cCI6MTYxMTMyNTg2NSwianRpIjoiYjJhNDY5YTItZGE1My00MGM1LWExNDYtMDg1ZTIxNzBkOTYwIiwiaWQiOjEsInJscyI6IiIsInJmX2V4cCI6MTYxMzgzMTQ2NX0.s702zGzynnTmyBgc0VzB6HJvccIRPMrH75pQ0YcR0fY',
  },
  // credentials: 'true',
};

/**
 * Represents everythin API related
 */
export default class API {
  /**
   * Does get requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @return {object} JSON response from API request
   */
  static get(route: string) {
    return axios
      .get(baseUrl() + route, axiosConfig)
      .then((res: AxiosResponse) => {
        return res.data;
      })
      .catch((error: AxiosError) => {
        console.log(error);
      });
  }

  /**
   * Does post requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @param {object} data - data to pass to the API
   * @return {object} JSON response from API request
   */
  static post(route: string, data: object) {
    return axios
      .post(baseUrl() + route, data, axiosConfig)
      .then((res: AxiosResponse) => {
        return res.data;
      })
      .catch((error: AxiosError) => {
        console.log(error);
      });
  }
}
