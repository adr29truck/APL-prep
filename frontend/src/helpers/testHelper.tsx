import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

export const TestHelper = ({ children }: any) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};
