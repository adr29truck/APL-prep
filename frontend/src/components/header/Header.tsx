import React, { Fragment } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import './Header.css';
import Typography from '@material-ui/core/Typography';
import ArrowLeftIcon from '@material-ui/icons/ArrowLeft';
import ArrowRightIcon from '@material-ui/icons/ArrowRight';

import { getCurrentTime } from '../../features/time/selectors';

const Header = () => {
  const dispatch = useDispatch();
  const day = useSelector(getCurrentTime);

  function onClickLeftArrow() {
    dispatch({ type: 'time/reduceTime' });
  }
  function onClickRightArrow() {
    dispatch({ type: 'time/increaseTime' });
  }

  return (
    <Fragment>
      <header>
        <ArrowLeftIcon onClick={onClickLeftArrow} />
        <h3 className="text-center">Timeline</h3>
        <ArrowRightIcon onClick={onClickRightArrow} />
      </header>
      <Typography
        variant="h2"
        component="h2"
        gutterBottom
        className="text-center"
      >
        {day}
      </Typography>
    </Fragment>
  );
};

export default Header;
