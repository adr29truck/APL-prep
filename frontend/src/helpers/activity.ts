import API from './API';
import store from '../store';

export async function fetchActivities() {
  const temp = await API.get('activities');
  store.dispatch({ type: 'activity/setActivitiesState', payload: temp });
}
