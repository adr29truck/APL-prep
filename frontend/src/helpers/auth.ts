import store from '../store';
import API from './API';

/**
 * Extracts the content from a JWT token
 * @param {*} token - x
 * @return {Object} - data
 */
export function parseJwt(token: string) {
  const base64Url = token.split('.')[1];
  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
  const jsonPayload = decodeURIComponent(
    atob(base64)
      .split('')
      .map((c) => {
        return `%${`00${c.charCodeAt(0).toString(16)}`.slice(-2)}`;
      })
      .join('')
  );
  return JSON.parse(jsonPayload);
}

export async function signIn(payload: object) {
  const temp = await API.post('api/login', payload);
  store.dispatch({ type: 'auth/setState', payload: temp });
}

export async function signOut() {
  await API.post('api/logout');
  store.dispatch({ type: 'auth/setState', payload: { id: 0 } });
}
