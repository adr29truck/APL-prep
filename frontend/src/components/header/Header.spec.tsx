import React from 'react';
import { mount } from 'enzyme';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import '@testing-library/jest-dom/extend-expect';
import Header from './Header';
import { TestHelper } from '../../helpers/testHelper';

// Must mock the date class else it would fail the snapshots
class MockDate extends Date {
  constructor() {
    super('2020-05-14T11:01:58.135Z');
  }
}
// @ts-ignore
global.Date = MockDate;

describe('<Header />', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <TestHelper>
        <Header />
      </TestHelper>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('handles right click correctly', () => {
    const temp = new Date()
    const date = temp.getDate()
    const today = date < 10 ? '0'+date : date
    const tomorrow = date+1 < 10 ? '0'+date+1 : date+1
    render(
      <TestHelper>
        <Header />
      </TestHelper>
    );
    expect(screen.getByText(today, {exact: false}))
    userEvent.click(screen.getByTestId('right-arrow'));
    userEvent.click(screen.getByTestId('left-arrow'));
  });
});
