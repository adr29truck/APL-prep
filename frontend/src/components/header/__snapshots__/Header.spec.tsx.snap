// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `"<header class=\\"center\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"left-arrow\\"><path d=\\"M14 7l-5 5 5 5V7z\\"></path></svg><h3 class=\\"text-center\\">Timeline</h3><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"right-arrow\\"><path d=\\"M10 17l5-5-5-5v10z\\"></path></svg></header><h2 class=\\"MuiTypography-root text-center MuiTypography-h2 MuiTypography-gutterBottom\\">2020-05-14</h2>"`;
